src/App.js

```diff
@@ -7,42 +7,73 @@ import { Profile } from './components/Profile';
-//import Registration from './components/UserPathLog/Register';
-//import Resetpw from './components/UserPathLog/Resetpw';
-//import { HashRouter } from 'react-router-dom';
-
-const App = () => {
-  return (
-    <Router>
-      <Switch>
-        {/*<a className="logoref" href="http://localhost:3000/welcome#/"> </a>*/}
-        <PrivateRoute path="/dashboard">
-          <Dashboard />
-        </PrivateRoute>
-        <PrivateRoute path="/profile">
-          <Profile />
-        </PrivateRoute>
-
-        <Route path="/findpeople">
-          <FindPeople />
-        </Route>
-        <Route path="/user/:id" component={OtherProfile} />
-
-        <Route path="/findpeople">
-          <FindPeople />
-        </Route>
-        {/* <HashRouter> */}
-        <Route path="/">
-          <Login />
-        </Route>
-        {/* <Route exact path="/register" component={Registration}> Register </Route>
-				<Route exact path="/resetpw" component={Resetpw}> Reset PW</Route> */}
-        {/* </HashRouter> */}
-        {/* add Register and Reset PW */}
-        {/* <Route path="/user/:id" component={OtherProfile}></Route> */}
-      </Switch>
-    </Router>
-  );
+import Registration from './components/UserPathLog/Register';
+import Resetpw from './components/UserPathLog/Resetpw';
+import Friends from './components/Friends/Friends.js'
+import { HashRouter, Link } from "react-router-dom";
+
+// import Profile from './components/Profile'
+// import { TestLogin } from './components/TestLogin/TestLogin';
+// import { CirclesBackground } from "./components/styleElements/CirclesBackground";
+// import { BottomNav } from "./components/BottomNav";
+
+
+const App = (props) => {
+	console.log("props in app", props)
+	const [otherProfileID, setOtherProfileID] = useState({})
+	const [IDOther, setIDOther] = useState({})
+
+	const handleProfileID = (newAppID) => {
+		// console.log("otherprofileID", typeof (otherProfilID))
+		setOtherProfileID(newAppID)
+		// console.log("props in handlechange", setIDOther(newAppID))
+		// props.onChange(newAppID)
+		// props.onChange(newAppID)
+	}
+
+	const sendToOtherProfile = (newAppID) => {
+		console.log("I am the visited profile ID in APP", newAppID)
+
+		setIDOther(newAppID)
+		props.onChange(newAppID)
+
+
+	}
+
+	return (
+		<Router>
+			<Switch>
+				<Route exact path="/" >
+					<Login />
+				</Route>
+
+				<Route exact path="/register">
+					<Registration />
+				</Route>
+				<Route exact path="/resetpw">
+					<Resetpw />
+				</Route>
+				<PrivateRoute path='/dashboard'>
+					<Dashboard />
+				</PrivateRoute>
+				{/* <PrivateRoute path='/profile'>
+					<Profile />
+				</PrivateRoute> */}
+
+				<Route exact path='/findpeople'>
+					<FindPeople />
+				</Route>
+
+				<Route exact path='/friends'>
+					<Friends name={otherProfileID} onChange={handleProfileID} />
+				</Route>
+				<Route exact path='/user/:id'>
+					<OtherProfile name={IDOther} onChange={sendToOtherProfile} />
```

Comment:
I'm fine if you know what you're doing here.  I assume this is the bit where you find people and look at their profiles and invite them to be friends.  But wouldn't we want that functionality be in a privateRoute?  And how is it going to integrated into the app?

Reply:
this part is not yet functioning fully as I have a hard time passing the data from app to OtherProfile. What do you mean about how is it going to integrate to the app?